app/javascript/components/lesson-preview/components/share-button.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import axios from '../../../src/js/axiosWithCsrf';
+
+const ShareButton = ({text, classes, onClick}) => {
+  const [copied, setCopied] = useState(false);
+
+  const handleOnClick = (event) => {
+    onClick(event.target.value);
+  };
+
+  return (
+    <button type="button" className={`button ${classes}`} onClick={handleOnClick} >
+      {text}
```

Comment:
Nit: can we use the copied conditional here instead of passing the text in as a prop please?
```suggestion
      {copied ? 'Copied!' : 'Share'}
```

All the copy/share logic can be within this component instead of the parent. The only thing I think we'll need to pass in as a prop is the content.